src/main/java/com/fasterxml/jackson/core/sym/CharsToNameCanonicalizer.java

```diff
@@ -114,6 +114,11 @@ public final class CharsToNameCanonicalizer
      */
     final protected AtomicReference<TableInfo> _tableInfo;
 
+    /**
+     * @since 2.16
+     */
+    final protected JsonFactory _jsonFactory;
```

Comment:
Ah sorry, I should have been bit clearer here. I think that `JsonFactory` (or ideally just `TokenStreamFactory` if it has all access) should be passed to factory methods, but only used to access configuration that is needed; that configuration (`StreamReadConstraints`, anything else) should be retained but not factory.